var/spack/repos/builtin/packages/caliper/package.py

```diff
@@ -24,27 +24,61 @@
 ##############################################################################
 from spack import *
 
-
-class Caliper(Package):
-    """Caliper is a generic context annotation system. It gives programmers the
-    ability to provide arbitrary program context information to (performance)
-    tools at runtime.
+class Caliper(CMakePackage):
+    """Caliper is a program instrumentation and performance measurement
+    framework. It provides data collection mechanisms and a source-code
+    annotation API for a variety of performance engineering use cases,
+    e.g., performance profiling, tracing, monitoring, and
+    auto-tuning.
     """
 
     homepage = "https://github.com/LLNL/Caliper"
     url      = ""
 
     version('master', git='https://github.com/LLNL/Caliper.git')
 
-    variant('mpi', default=True, description='Enable MPI function wrappers.')
+    variant('mpi', default=True, 
+            description='Enable MPI wrappers')
+    variant('dyninst', default=False, 
+            description='Enable symbol translation support with dyninst')
+    variant('callpath', default=True,
+            description='Enable callpath service (requires libunwind)')
+    variant('papi', default=True, description='Enable PAPI service')
+    variant('gotcha', default=True, description='Enable GOTCHA support')
 
-    depends_on('libunwind')
-    depends_on('papi')
+    depends_on('dyninst', when='+dyninst')
+    depends_on('papi', when='+papi')
     depends_on('mpi', when='+mpi')
+    depends_on('libunwind', when='+callpath')
+
     depends_on('cmake', type='build')
+    depends_on('python', type='build')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        args = [
+            '-DBUILD_TESTING=Off',
+            '-DWITH_DOCS=Off',
+            '-DWITH_TEST_APPS=Off'
+        ]
+
+        if '+dyninst' not in spec:
+            args.append('-DWITH_DYNINST=Off')
+        if '+callpath' not in spec:
+            args.append('-DWITH_CALLPATH=Off')
+        if '+gotcha' not in spec:
+            args.append('-DWITH_GOTCHA=Off')
```

Comment:
For variants like this, you should _explicitly enable_ the flag `if '+variant' in spec` and _explicitly disable_ the flag otherwise.